src/backend/mailbox_manager/tests/api/mail_domain_accesses/test_api_mail_domains_accesses_get_available_users.py

```diff
@@ -0,0 +1,169 @@
+"""
+Tests for MailDomainAccess API endpoint in People's mailbox manager app.
+Focus on get available users.
+"""
+
+import pytest
+from rest_framework import status
+from rest_framework.test import APIClient
+
+from core import factories as core_factories
+from core import models as core_models
+
+from mailbox_manager import enums, factories
+
+pytestmark = pytest.mark.django_db
+
+
+def test_api_mail_domain__available_users_anonymous():
+    """Anonymous users should not be allowed to list users."""
+    maildomain = factories.MailDomainFactory()
+
+    response = APIClient().get(
+        f"/api/v1.0/mail-domains/{maildomain.slug}/accesses/users/"
+    )
+
+    assert response.status_code == status.HTTP_401_UNAUTHORIZED
+    assert response.json() == {
+        "detail": "Authentication credentials were not provided."
+    }
+
+
+def test_api_mail_domain__available_users_forbidden():
+    """Authenticated user without accesses on maildomain should not be able to see available
+    users.
+    """
+    authenticated_user = core_factories.UserFactory()
+    client = APIClient()
+    client.force_login(authenticated_user)
+    maildomain = factories.MailDomainFactory()
+
+    response = client.get(f"/api/v1.0/mail-domains/{maildomain.slug}/accesses/users/")
+
+    assert response.status_code == status.HTTP_403_FORBIDDEN
+
+
+@pytest.mark.parametrize(
+    "role",
+    [
+        enums.MailDomainRoleChoices.OWNER,
+        enums.MailDomainRoleChoices.ADMIN,
+        enums.MailDomainRoleChoices.VIEWER,
+    ],
+)
+def test_api_mail_domain__list_available_users__with_abilities(role):
+    """Authenticated users with roles owner and admin should be allowed to list available users
+    for a domain.
+    """
+    dave = core_factories.UserFactory(email="bowbow@example.com", name="David Bowman")
+    nicole = core_factories.UserFactory(
+        email="nicole_foole@example.com", name="Nicole Foole"
+    )
+    frank = core_factories.UserFactory(
+        email="frank_poole@example.com", name="Frank Poole"
+    )
+    mary = core_factories.UserFactory(email="mary_pol@example.com", name="Mary Pol")
+
+    expected_ids = {str(user.id) for user in core_models.User.objects.all()}
+
+    authenticated_user = core_factories.UserFactory(name="Owen Rights")
+    client = APIClient()
+    client.force_login(authenticated_user)
+
+    maildomain = factories.MailDomainFactory(name="example.com")
+    factories.MailDomainAccessFactory(
+        user=authenticated_user,
+        domain=maildomain,
+        role=role,
+    )
+
+    response = client.get(f"/api/v1.0/mail-domains/{maildomain.slug}/accesses/users/")
+    if role == enums.MailDomainRoleChoices.VIEWER:
+        assert response.status_code == status.HTTP_403_FORBIDDEN
+    else:
```

Comment:
I'm sorry, but this is a bad pattern in tests, there should not be logic like that here :/